examples/react/src/App.tsx

```diff
@@ -1,74 +1,42 @@
-import CohortSDK from '@cohort-xyz/cohort-embed-js';
-import {useCallback, useEffect, useState} from 'react';
-
-function App() {
-  const [sdk, setSdk] = useState<CohortSDK | null>(null);
-
-  useEffect(() => {
-    const instance = new CohortSDK(import.meta.env.COHORT_XPS_ORIGIN_URL, true);
-
-    const offLocation = instance.on('location.updated', message => {
-      console.log('Location updated', message);
-    });
-
-    const offOrder = instance.on('order.created', message => {
-      console.log('Order created', message);
-    });
-
-    setSdk(instance);
-    return () => {
-      offLocation();
-      offOrder();
-      instance.destroy();
-      setSdk(null);
-    };
-  }, []);
-
-  const renderExperienceSpace = useCallback(
-    (node: HTMLElement | null) => {
-      if (node && sdk) {
-        const AUTH_TOKEN = import.meta.env.COHORT_AUTH_TOKEN;
-
-        sdk.renderExperienceSpace(
-          import.meta.env.COHORT_USER_EMAIL,
-          {
-            container: node,
-            // pathname: '/rewards',
-            // spinnerStyle: {
-            // 	backgroundColor: 'black',
-            // 	color: 'white',
-            // },
-            // iframeStyle: {
-            // 	width: '400px',
-            // },
-            // urlParams: {
-            // 	navigationType: 'burger',
-            // 	navbar: false,
-            // },
-          },
-          async () => {
-            // This is a mock function to simulate the call to get the authentication token
-            return new Promise(resolve => {
-              setTimeout(() => {
-                resolve(AUTH_TOKEN);
-              }, 1000);
-            });
-          },
-        );
-      }
-    },
-    [sdk],
-  );
+import {useState} from 'react';
+import {BrowserRouter as Router, Route, Routes} from 'react-router-dom';
+import Login from './Login';
+import ExperienceSpace from './ExperienceSpaceWithCustomLogin';
+import ExperienceSpaceWithCohortLogin from './ExperienceSpaceWithCohortLogin';
+
+const App = () => {
+  const [userEmail, setUserEmail] = useState<string | null>(localStorage.getItem('userEmail'));
+
+  const handleLogin = (email: string) => {
+    setUserEmail(email);
+    localStorage.setItem('userEmail', email);
+    console.log('[PARENT] Logged in as', email);
+    // If there is a redirect param in the URL, redirect to that URL
+    const urlParams = new URLSearchParams(window.location.search);
+    const destination = urlParams.get('redirectUri');
+    if (destination) {
+      console.log('[PARENT] Redirecting to', destination);
+      window.location.href = destination;
+    }
+  };
+  const handleLogout = () => {
+    setUserEmail(null);
+    localStorage.removeItem('userEmail');
+    console.log('[PARENT] Logged out');
+  };
 
   return (
-    <div
-      style={{
-        width: '100%',
-        height: '100%',
-      }}
-      ref={renderExperienceSpace}
-    />
+    <Router>
+      <Routes>
```

Comment:
You haven't setup a default route therefore when launching the demo app you have a white page.

I think you should have a default page that serves as a menu to avoid people to have to look inside the router to know which route they can try

Reply:
Is it ok if I let you do that?